chain/chain/src/state_sync.rs

```diff
@@ -56,23 +56,20 @@ fn save_epoch_new_chunks<T: ChainStoreAccess>(
         return Ok(());
     }
 
-    let mut done = true;
-    for (num_new_chunks, new_chunk) in num_new_chunks.iter_mut().zip(header.chunk_mask().iter()) {
-        // Only need to reach 2, so don't bother adding more than that
-        if *new_chunk && *num_new_chunks < 2 {
-            *num_new_chunks += 1;
-        }
-        if *num_new_chunks < 2 {
-            done = false;
-        }
-    }
+    let done = num_new_chunks.iter().all(|num_chunks| *num_chunks >= 2);
```

Comment:
Ahh was this a bug before too? So this change will push out the sync hash one more block (with no missed chunks, 4th block of the epoch instead of the 3rd). Was the previous one incorrect somewhere? The logic in that state sync header generation is a bit tricky, could def have missed something there

Reply:
Oh nvm I see your msg on zulip, yea this is a good fix, thanks!

I think this was never caught because the `state_sync.rs` tests [set up skipped chunks](https://github.com/near/nearcore/blob/97f69fe08ee4744072374d2a96cb8db9da31afcf/integration-tests/src/test_loop/tests/state_sync.rs#L143), but this wont be hit there bc theres no resharding. And the resharding shard shuffling test doesnt skip chunks. So maybe would be good to add some resharding tests w/ `drop_chunks_by_height()` or something